include/dxc/DXIL/DxilShaderFlags.h

```diff
@@ -359,7 +368,9 @@ class ShaderFlags {
   unsigned m_bRequiresGroup : 1; // SHADER_FEATURE_OPT_REQUIRES_GROUP
                                  // (OptFeatureInfo_RequiresGroup)
 
-  uint32_t m_align1 : 23; // align to 64 bit.
+  unsigned m_UnusedResourceBinding : 3;
```

Comment:
I don't think we should be adding this setting into the `ShaderFlags`. These are visible to the driver and part of the DIXL contract, so we must be careful when adding flags here.

Instead, we need something like a LLVM module flag, which is stripped from the module when stripping reflection. It should be serialized along with HLModule and DxilModule metadata.

We already have metadata serialization for `LegacyResourceReservation`, through HLOptions serialization for HLModule and `IntermediateOptions` serialization in `DxilModule`. This is the right place for this option as well. The one issue is preservation of a flag indicating preservation of  `IntermediateOptions`